standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -1039,8 +1136,27 @@ public static ConfVars getMetaConf(String name) {
             "More users can be added in ADMIN role later."),
     USE_SSL("metastore.use.SSL", "hive.metastore.use.SSL", false,
         "Set this to true for using SSL encryption in HMS server."),
+    // We should somehow unify next two options.
     USE_THRIFT_SASL("metastore.sasl.enabled", "hive.metastore.sasl.enabled", false,
         "If true, the metastore Thrift interface will be secured with SASL. Clients must authenticate with Kerberos."),
+    METASTORE_CLIENT_USE_PLAIN_AUTH("metastore.client.use.plain.auth",
+            "metastore.client.use.plain.auth", false,
+            "If true, clients will authenticate using plain authentication, by providing username" +
+                    " and password."),
+    METASTORE_CLIENT_PLAIN_USERNAME("metastore.client.plain.username",
+            "metastore.client.plain.username",  "",
+        "The username used by the metastore client when " +
+                    METASTORE_CLIENT_USE_PLAIN_AUTH + " is true. The password is obtained from " +
+                    CredentialProviderFactory.CREDENTIAL_PROVIDER_PATH + " using username as the " +
+                    "alias."),
+    THRIFT_AUTH_CONFIG_USERNAME("metastore.authentication.config.username",
```

Comment:
The first one is client side config. It will be used whenever there's PLAIN authentication. That specifies what username to provide to the server.

The other is server side and used only when CONFIG authentication is used. It's the username that will be matched against the username provided by the client.